### PR TITLE
[d16-8][watchOS] Add x86_64 simulator support (#10059)

### PR DESCRIPTION
**Pull request by dalexsoto**
Fixes https://github.com/xamarin/xamarin-macios/issues/9949

> watchOS apps don't deploy to watchOS 7+ simulators

* Bumps mono binaries to include x86_64 watchOS support
* Build runtime/registrar x86_64 slices
* Produce a 64 bit version of Xamarin.WatchOS.dll
* Allow building x86_64 for watch simulators in mtouch
* Let xharness know about x86_64

* [tests] Add x86_64 arch to test-libraries

* Make dotnet package aware of x64

* [ObjCRuntime] Fix computing if we're calling a stret function or not in a 64-bit watchOS simulator.

* [xharness] Re-enable some watchOS tests.

Co-authored-by: Rolf Bjarne Kvinge <rolf@xamarin.com>

New commits in mono/mono:

* mono/mono@ac596375c76 Add support for OP_FCONV_TO_I to mini-arm64.c. (#20548)
* mono/mono@392fe5b87b2 [2020-02][watchOS] Add simwatch64 support (#20552)
* mono/mono@a22ed3f094e Fix potential crash for Encoder.Convert (#20522)
* mono/mono@970783731fc Bump to F# 5.0 (#20511)
* mono/mono@32ab5066f72 Bump msbuild to fix a build issue
* mono/mono@93a7fe77e8e Ensure special static slots respect alignment. (#20506)
* mono/mono@3db5b358413 [debugger] Switch to GC Unsafe in signal handler callbacks (#20495)
* mono/mono@af315f44c40 [2020-02][corlib] ThreadAbortException protection for ArraySortHelper (#20468)
* mono/mono@ca11fb0fd81 [2020-02] Bump ikvm-fork to include https://github.com/mono/ikvm-fork/pull/20 (#20452)

Diff: https://github.com/mono/mono/compare/be2226b5a1c57df065efc4c1cf008d581e5cec7d..ac596375c762c6b8dbe3c802f0ce626004eab51c

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -821,6 +821,19 @@ $(WATCHOS_DOTNET_BUILD_DIR)/32/Xamarin.WatchOS%dll $(WATCHOS_DOTNET_BUILD_DIR)/3
 		$(WATCHOS_SOURCES) \
 		@$<
 
+$(WATCHOS_DOTNET_BUILD_DIR)/64/Xamarin.WatchOS%dll $(WATCHOS_DOTNET_BUILD_DIR)/64/Xamarin.WatchOS%pdb $(WATCHOS_DOTNET_BUILD_DIR)/ref/Xamarin.WatchOS%dll: $(WATCHOS_DOTNET_BUILD_DIR)/watchos-generated-sources $(WATCHOS_SOURCES) $(PRODUCT_KEY_PATH) | $(WATCHOS_DOTNET_BUILD_DIR)/64 $(WATCHOS_DOTNET_BUILD_DIR)/ref
+	$(Q_DOTNET_GEN) \
+		$(SYSTEM_CSC) $(DOTNET_FLAGS) -out:$(WATCHOS_DOTNET_BUILD_DIR)/64/Xamarin.WatchOS.dll -optimize \
+		-publicsign -keyfile:$(PRODUCT_KEY_PATH) \
+		-refout:$(WATCHOS_DOTNET_BUILD_DIR)/ref/Xamarin.WatchOS.dll \
```
Comment (spouliot) [inline src/Makefile:828]: you should pick up Rolf's PR to remove the dupe (from `/32/`)

**Review by spouliot — APPROVED** [review]

**Comment by dalexsoto** [comment]
Will cherry-pick once @rolfbjarne merges it into main

**Comment by spouliot** [comment]
@dalexsoto `main` seems to have watchOS related failures, better hold on the backport until those are fixed

File: mk/rules.mk [hunk 2]
```diff
@@ -91,19 +92,34 @@ define NativeCompilationTemplate
 ## watch simulator
 
 .libs/watchsimulator/%$(1).x86.o: %.m $(EXTRA_DEPENDENCIES) | .libs/watchsimulator
-	$$(call Q_2,OBJC,  [watchsimulator]) $(SIMULATOR_CC) $(SIMULATORWATCH_OBJC_CFLAGS) $$(EXTRA_DEFINES) $(SIMW_I) -g $(2) -c $$< -o $$@
+	$$(call Q_2,OBJC,  [watchsimulator]) $(SIMULATOR_CC) $(SIMULATORWATCH_OBJC_CFLAGS)   $$(EXTRA_DEFINES) $(SIMW_I) -g $(2) -c $$< -o $$@
```
Comment (mandel-macaque) [inline mk/rules.mk:95]: Why the space chang?

**Review by mandel-macaque — APPROVED** [review]
I don't understand the addition of noise.

**Review by tj-devel709 — APPROVED** [review]

**Comment by dalexsoto** [comment]
Moving to xcode12.3

**Comment by appsure** [comment]
Any ETA on when this will all be included in the next VS release (preview or stable)?

**Pull request closed without merging**